manual.typ

```diff
@@ -1415,3 +1374,187 @@ tree.tree(data, content: (padding: .1), direction: "right",
+  / min: Axis min. value
+  / max: Axis max. value
+  / ticks: List of tick values or value/label tuples
+  / tick-step: Tick step or `none`
+  / minor-tick-step: Minor tick step or `none`
+  / decimals: Number of tick label decimal digits
+  / unit: Tick label suffix
+  ]
+])
+#def-arg("body", "..",
+  [Calls of `plot.add(..)`, see <plot-add>])
+
+=== Plot-Add
+
+The `plot.add` function adds plotting data into a plot environment.
+It must be called from insides `plot({..})`.
+
+If used with a callback, the `domain` must be specified!
+
+```typc
+add(domain: auto, hypograph: false, epigraph: false, fill: false,
+    mark: none, mark-size: .2, mark-style: (:), samples: 100,
+    style: (:), axes: ("x", "y"),
+    data)
+```
+
+#def-arg("domain", "a|auto",
+  [Range of x for sampled data])
+#def-arg("hypograph", "b",
+  [Fill graph below function])
+#def-arg("epigraph", "b",
+  [Fill graph above function])
+#def-arg("fill", "b",
+  [Fill graph to zero])
+#def-arg("mark", "s?",
+  [Mark symbol. Any of `("x", "o", "square", "triangle", "|", "-")`])
+#def-arg("mark-size", "f?",
+  [Size of mark symbol])
+#def-arg("mark-style", "style?",
+  [Style used for drawing marks. Note that this inherits the plots style])
+#def-arg("style", "style",
+  [Style used for drawing the graph])
+#def-arg("samples", "i",
+  [Number of samples to use (only if `data` is of type function)])
+#def-arg("axes", "a",
+  [Array of axis names to use])
+#def-arg("data", "function|a",
+  [Array of data points or function in the form `x => y`])
+
+#example({
+  import "draw.typ": *
+  plot.plot(size: (2,2), x-tick-step: 180, y-tick-step: 1,
+                         x-unit: $degree$, {
+    plot.add(domain: (0, 360), x => calc.sin(x * 1deg))
+  })
+}, ```typc
+plot.plot(size: (2,2), x-tick-step: 180, y-tick-step: 1,
+          x-unit: $degree$, {
+  plot.add(domain: (0, 360), x => calc.sin(x * 1deg))
+})
+```)
+
+#example({
+  import "draw.typ": *
+  plot.plot(size: (2,2), x-tick-step: 180, y-tick-step: 1,
+                         x-unit: $degree$, y-max: .5, {
+    plot.add(domain: (0, 360), x => calc.sin(x * 1deg))
+    plot.add(domain: (0, 360), x => calc.cos(x * 1deg),
+             samples: 10, mark: "x", mark-style: (stroke: blue))
+  })
+}, ```typc
+plot.plot(size: (2,2), x-tick-step: 180, y-tick-step: 1,
+          x-unit: $degree$, y-max: .5, {
+  plot.add(domain: (0, 360), x => calc.sin(x * 1deg))
+  plot.add(domain: (0, 360), x => calc.cos(x * 1deg),
+           samples: 10, mark: "x", mark-style: (stroke: blue))
+})
+```)
+
+== Chart
+
+With the `chart` library it is easy to draw charts.
+
+=== Barchart
+
+```typc
+barchart(size: (width, height))
+```
+
+#def-arg("data", "a",
+  [Data array of arrays or dictionaries])
+#def-arg("label-key", "s|i",
+  [Data item key to access an items label])
+#def-arg("value-key", "s|i|a",
+  [Data item key(s) to access an items value(s). For multi-value charts
+   this must be an array of all keys, e.G. `(..range(1, 5))`])
```

Comment:
I think the format of the data needs a bit more information as the label and value keys don't make sense.